[TEST-FAILURE] map.ClientMapPartitionLostListenerTest.test_mapPartitionLostListener_invoked_fromOtherNode

```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at org.junit.Assert.assertFalse(Assert.java:74)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest$1.run(ClientMapPartitionLostListenerTest.java:97)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:737)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:751)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest.assertMapPartitionLostEventEventually(ClientMapPartitionLostListenerTest.java:91)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest.test_mapPartitionLostListener_invoked_fromOtherNode(ClientMapPartitionLostListenerTest.java:124)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-IbmJDK1.6/com.hazelcast$hazelcast-client-new/285/testReport/junit/com.hazelcast.client.map/ClientMapPartitionLostListenerTest/test_mapPartitionLostListener_invoked_fromOtherNode/

--- a/hazelcast-client-new/src/test/java/com/hazelcast/client/map/ClientMapPartitionLostListenerTest.java
+++ b/hazelcast-client-new/src/test/java/com/hazelcast/client/map/ClientMapPartitionLostListenerTest.java
@@ -1,5 +1,7 @@
 package com.hazelcast.client.map;
 
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.impl.HazelcastClientInstanceImpl;
 import com.hazelcast.client.test.TestHazelcastFactory;
 import com.hazelcast.config.Config;
 import com.hazelcast.core.HazelcastInstance;
@@ -7,6 +9,7 @@ import com.hazelcast.map.MapPartitionLostEvent;
 import com.hazelcast.map.MapPartitionLostListenerStressTest.EventCollectingMapPartitionLostListener;
 import com.hazelcast.map.impl.MapService;
 import com.hazelcast.map.listener.MapPartitionLostListener;
+import com.hazelcast.nio.Address;
 import com.hazelcast.partition.InternalPartitionLostEvent;
 import com.hazelcast.spi.EventRegistration;
 import com.hazelcast.spi.impl.eventservice.InternalEventService;
@@ -22,8 +25,10 @@ import org.junit.runner.RunWith;
 import java.util.Collection;
 import java.util.List;
 
+import static com.hazelcast.client.impl.ClientTestUtil.getHazelcastClientInstanceImpl;
 import static com.hazelcast.map.impl.MapService.SERVICE_NAME;
 import static com.hazelcast.test.HazelcastTestSupport.assertTrueEventually;
+import static com.hazelcast.test.HazelcastTestSupport.getAddress;
 import static com.hazelcast.test.HazelcastTestSupport.getNode;
 import static com.hazelcast.test.HazelcastTestSupport.randomMapName;
 import static org.junit.Assert.assertEquals;
@@ -110,7 +115,14 @@ public class ClientMapPartitionLostListenerTest {
 
         final HazelcastInstance instance1 = hazelcastFactory.newHazelcastInstance(config);
         final HazelcastInstance instance2 = hazelcastFactory.newHazelcastInstance(config);
-        final HazelcastInstance client = hazelcastFactory.newHazelcastClient();
+        final ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getNetworkConfig().setSmartRouting(false);
+        final HazelcastInstance client = hazelcastFactory.newHazelcastClient(clientConfig);
+
+        final HazelcastClientInstanceImpl clientInstanceImpl = getHazelcastClientInstanceImpl(client);
+        final Address clientOwnerAddress = clientInstanceImpl.getClientClusterService().getOwnerConnectionAddress();
+
+        final HazelcastInstance other = getAddress(instance1).equals(clientOwnerAddress) ? instance2 : instance1;
 
         final EventCollectingMapPartitionLostListener listener = new EventCollectingMapPartitionLostListener(0);
         client.getMap(mapName).addPartitionLostListener(listener);
@@ -118,7 +130,7 @@ public class ClientMapPartitionLostListenerTest {
         assertRegistrationsSizeEventually(instance1, mapName, 1);
         assertRegistrationsSizeEventually(instance2, mapName, 1);
 
-        final MapService mapService = getNode(instance2).getNodeEngine().getService(SERVICE_NAME);
+        final MapService mapService = getNode(other).getNodeEngine().getService(SERVICE_NAME);
         final int partitionId = 5;
         mapService.onPartitionLost(new InternalPartitionLostEvent(partitionId, 0, null));
 

--- a/hazelcast-client-new/src/test/java/com/hazelcast/client/partitionservice/ClientPartitionLostListenerTest.java
+++ b/hazelcast-client-new/src/test/java/com/hazelcast/client/partitionservice/ClientPartitionLostListenerTest.java
@@ -1,5 +1,7 @@
 package com.hazelcast.client.partitionservice;
 
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.impl.HazelcastClientInstanceImpl;
 import com.hazelcast.client.test.TestHazelcastFactory;
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.nio.Address;
@@ -28,9 +30,11 @@ import java.io.IOException;
 import java.util.Collection;
 import java.util.List;
 
+import static com.hazelcast.client.impl.ClientTestUtil.getHazelcastClientInstanceImpl;
 import static com.hazelcast.partition.InternalPartitionService.PARTITION_LOST_EVENT_TOPIC;
 import static com.hazelcast.partition.InternalPartitionService.SERVICE_NAME;
 import static com.hazelcast.test.HazelcastTestSupport.assertTrueEventually;
+import static com.hazelcast.test.HazelcastTestSupport.getAddress;
 import static com.hazelcast.test.HazelcastTestSupport.getNode;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
@@ -93,7 +97,14 @@ public class ClientPartitionLostListenerTest {
     public void test_partitionLostListener_invoked_fromOtherNode() {
         final HazelcastInstance instance1 = hazelcastFactory.newHazelcastInstance();
         final HazelcastInstance instance2 = hazelcastFactory.newHazelcastInstance();
-        final HazelcastInstance client = hazelcastFactory.newHazelcastClient();
+        final ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getNetworkConfig().setSmartRouting(false);
+        final HazelcastInstance client = hazelcastFactory.newHazelcastClient(clientConfig);
+
+        final HazelcastClientInstanceImpl clientInstanceImpl = getHazelcastClientInstanceImpl(client);
+        final Address clientOwnerAddress = clientInstanceImpl.getClientClusterService().getOwnerConnectionAddress();
+
+        final HazelcastInstance other = getAddress(instance1).equals(clientOwnerAddress) ? instance2 : instance1;
 
         final EventCollectingPartitionLostListener listener = new EventCollectingPartitionLostListener();
         client.getPartitionService().addPartitionLostListener(listener);
@@ -101,7 +112,7 @@ public class ClientPartitionLostListenerTest {
         assertRegistrationsSizeEventually(instance1, 1);
         assertRegistrationsSizeEventually(instance2, 1);
 
-        final InternalPartitionServiceImpl partitionService = getNode(instance2).getNodeEngine().getService(SERVICE_NAME);
+        final InternalPartitionServiceImpl partitionService = getNode(other).getNodeEngine().getService(SERVICE_NAME);
         final int partitionId = 5;
         partitionService.onPartitionLost(new InternalPartitionLostEvent(partitionId, 0, null));
 

--- a/hazelcast-client/src/test/java/com/hazelcast/client/map/ClientMapPartitionLostListenerTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/map/ClientMapPartitionLostListenerTest.java
@@ -1,5 +1,7 @@
 package com.hazelcast.client.map;
 
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.impl.HazelcastClientInstanceImpl;
 import com.hazelcast.client.test.TestHazelcastFactory;
 import com.hazelcast.config.Config;
 import com.hazelcast.core.HazelcastInstance;
@@ -7,6 +9,7 @@ import com.hazelcast.map.MapPartitionLostEvent;
 import com.hazelcast.map.MapPartitionLostListenerStressTest.EventCollectingMapPartitionLostListener;
 import com.hazelcast.map.impl.MapService;
 import com.hazelcast.map.listener.MapPartitionLostListener;
+import com.hazelcast.nio.Address;
 import com.hazelcast.partition.InternalPartitionLostEvent;
 import com.hazelcast.spi.EventRegistration;
 import com.hazelcast.spi.impl.eventservice.InternalEventService;
@@ -22,8 +25,10 @@ import org.junit.runner.RunWith;
 import java.util.Collection;
 import java.util.List;
 
+import static com.hazelcast.client.impl.ClientTestUtil.getHazelcastClientInstanceImpl;
 import static com.hazelcast.map.impl.MapService.SERVICE_NAME;
 import static com.hazelcast.test.HazelcastTestSupport.assertTrueEventually;
+import static com.hazelcast.test.HazelcastTestSupport.getAddress;
 import static com.hazelcast.test.HazelcastTestSupport.getNode;
 import static com.hazelcast.test.HazelcastTestSupport.randomMapName;
 import static org.junit.Assert.assertEquals;
@@ -110,7 +115,14 @@ public class ClientMapPartitionLostListenerTest {
 
         final HazelcastInstance instance1 = hazelcastFactory.newHazelcastInstance(config);
         final HazelcastInstance instance2 = hazelcastFactory.newHazelcastInstance(config);
-        final HazelcastInstance client = hazelcastFactory.newHazelcastClient();
+        final ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getNetworkConfig().setSmartRouting(false);
+        final HazelcastInstance client = hazelcastFactory.newHazelcastClient(clientConfig);
+
+        final HazelcastClientInstanceImpl clientInstanceImpl = getHazelcastClientInstanceImpl(client);
+        final Address clientOwnerAddress = clientInstanceImpl.getClientClusterService().getOwnerConnectionAddress();
+
+        final HazelcastInstance other = getAddress(instance1).equals(clientOwnerAddress) ? instance2 : instance1;
 
         final EventCollectingMapPartitionLostListener listener = new EventCollectingMapPartitionLostListener(0);
         client.getMap(mapName).addPartitionLostListener(listener);
@@ -118,7 +130,7 @@ public class ClientMapPartitionLostListenerTest {
         assertRegistrationsSizeEventually(instance1, mapName, 1);
         assertRegistrationsSizeEventually(instance2, mapName, 1);
 
-        final MapService mapService = getNode(instance2).getNodeEngine().getService(SERVICE_NAME);
+        final MapService mapService = getNode(other).getNodeEngine().getService(SERVICE_NAME);
         final int partitionId = 5;
         mapService.onPartitionLost(new InternalPartitionLostEvent(partitionId, 0, null));
 

--- a/hazelcast-client/src/test/java/com/hazelcast/client/partitionservice/ClientPartitionLostListenerTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/partitionservice/ClientPartitionLostListenerTest.java
@@ -1,5 +1,7 @@
 package com.hazelcast.client.partitionservice;
 
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.impl.HazelcastClientInstanceImpl;
 import com.hazelcast.client.test.TestHazelcastFactory;
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.nio.Address;
@@ -28,9 +30,11 @@ import java.io.IOException;
 import java.util.Collection;
 import java.util.List;
 
+import static com.hazelcast.client.impl.ClientTestUtil.getHazelcastClientInstanceImpl;
 import static com.hazelcast.partition.InternalPartitionService.PARTITION_LOST_EVENT_TOPIC;
 import static com.hazelcast.partition.InternalPartitionService.SERVICE_NAME;
 import static com.hazelcast.test.HazelcastTestSupport.assertTrueEventually;
+import static com.hazelcast.test.HazelcastTestSupport.getAddress;
 import static com.hazelcast.test.HazelcastTestSupport.getNode;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
@@ -93,7 +97,14 @@ public class ClientPartitionLostListenerTest {
     public void test_partitionLostListener_invoked_fromOtherNode() {
         final HazelcastInstance instance1 = hazelcastFactory.newHazelcastInstance();
         final HazelcastInstance instance2 = hazelcastFactory.newHazelcastInstance();
-        final HazelcastInstance client = hazelcastFactory.newHazelcastClient();
+        final ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getNetworkConfig().setSmartRouting(false);
+        final HazelcastInstance client = hazelcastFactory.newHazelcastClient(clientConfig);
+
+        final HazelcastClientInstanceImpl clientInstanceImpl = getHazelcastClientInstanceImpl(client);
+        final Address clientOwnerAddress = clientInstanceImpl.getClientClusterService().getOwnerConnectionAddress();
+
+        final HazelcastInstance other = getAddress(instance1).equals(clientOwnerAddress) ? instance2 : instance1;
 
         final EventCollectingPartitionLostListener listener = new EventCollectingPartitionLostListener();
         client.getPartitionService().addPartitionLostListener(listener);
@@ -101,7 +112,7 @@ public class ClientPartitionLostListenerTest {
         assertRegistrationsSizeEventually(instance1, 1);
         assertRegistrationsSizeEventually(instance2, 1);
 
-        final InternalPartitionServiceImpl partitionService = getNode(instance2).getNodeEngine().getService(SERVICE_NAME);
+        final InternalPartitionServiceImpl partitionService = getNode(other).getNodeEngine().getService(SERVICE_NAME);
         final int partitionId = 5;
         partitionService.onPartitionLost(new InternalPartitionLostEvent(partitionId, 0, null));
 

